non-critical-infra/packages/encrypt-email-address/encrypt-email-address.py

```diff
@@ -0,0 +1,100 @@
+#!/usr/bin/env python
+
+import re
+import click
+import subprocess
+from pathlib import Path
+from textwrap import dedent
+from textwrap import indent
+
+
+@click.command(
+    help=dedent("""\
+        Encrypt an email address (or email addresses) for inclusion in a mailing list.
+
+        See `non-critical-infra/modules/mailserver/mailing-lists.nix` for
+        documentation about how to use this.
+    """)
+)
+@click.argument("id")
+@click.argument("email")
+@click.option("--force/--no-force", "-f/ ", default=False)
+def main(id: str, email: str, force: bool):
+    # Feel free to make the regex less restrictive if you need to.
+    id_re = re.compile("[A-Za-z0-9-]+")
+    if not id_re.fullmatch(id):
+        raise click.ClickException(
+            f"Given ID: {id!r} is invalid. Must match regex: {id_re.pattern}"
+        )
+
+    # Make sure we aren't being given a text file that happens to have a newline at the end.
+    if "\n" in email:
+        raise click.ClickException(
+            "Email address must not not contain a newline character"
+        )
```

Comment:
This is kinda silly. Most editors append a final new line in linux land. Can't we rather strip whitespace?

Reply:
Isn't this email passed by cli rather than editor? But I would be also interested how we end up with a newline here.

Reply:
You're right @Mic92. The scenario I was vaguely worried about was somebody passing a file as a string, like this:

```
nix run .#encrypt-email-address jfly "$(< /tmp/foo)"
```

I'm also worried about the scenario where someone uses `sops edit` on these files directly, which will likely introduce a newline, and I can't think of anything clever we can do about it. Do y'all have any bright ideas?

Reply:
Mhm. Can we not just do a strip() instead to remove all whitespace around the string?

Reply:
Done.